### PR TITLE
Add section on PostgreSQL to the guides index [ci skip]

### PR DESCRIPTION
**Pull request by arbox**
### Summary

Add the already available section on Active Record and PostgreSQL to the guides index.

### Other Information

While searching for ActiveRecord datatypes I stumbled upon this guide: https://guides.rubyonrails.org/active_record_postgresql.html 

Since it is rather old (and complete), but still not listed on the main guides index I considered this a mistake. Please feel free to reject this PR if that section should not be included as part of the index.

### CODE REVIEW

File: guides/source/documents.yaml [hunk 1]
```diff
@@ -155,6 +155,10 @@
       name: Using Rails for API-only Applications
       url: api_app.html
       description: This guide explains how to effectively use Rails to develop a JSON API application.
+    -
+      name: Active Record and PostgreSQL
+      url: active_record_postgresql.html
```
Comment (rafaelfranca) [inline guides/source/documents.yaml]: Can you make it `work_in_progress: true`?

**Review by rafaelfranca — CHANGES_REQUESTED** [review]
This guide is work in progress and before making it to the index as finished we need to make it complete, following the guidelines and have the documentation team review.

File: guides/source/documents.yaml (hunk 1 above)
Comment (arbox) [reply guides/source/documents.yaml]: I've added a line to denote progress status.

**Comment by arbox** [comment]
@rafaelfranca 
Is it OK to add the guide on PostgreSQL into the advanced section?

**Comment by yskkin** [comment]
#23376 is related

**Comment by arbox** [comment]
I see, it's been a long story :)

**Pull request merged**